packages/checkout/widgets-lib/src/widgets/bridge/views/ClaimWithdrawal.tsx

```diff
@@ -0,0 +1,241 @@
+import { Box, Button, EllipsizedText } from '@biom3/react';
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { UserJourney, useAnalytics } from 'context/analytics-provider/SegmentAnalyticsProvider';
+import { useTranslation } from 'react-i18next';
+import { Transaction } from 'lib/clients';
+import { getChainNameById } from 'lib/chains';
+import { getL1ChainId } from 'lib';
+import { isPassportProvider } from 'lib/providerUtils';
+import { TransactionRequest } from '@ethersproject/providers';
+import { WalletProviderName } from '@imtbl/checkout-sdk';
+import { isNativeToken } from 'lib/utils';
+import { BigNumber } from 'ethers';
+import { SimpleLayout } from '../../../components/SimpleLayout/SimpleLayout';
+import { HeaderNavigation } from '../../../components/Header/HeaderNavigation';
+import { sendBridgeWidgetCloseEvent } from '../BridgeWidgetEvents';
+import { SimpleTextBody } from '../../../components/Body/SimpleTextBody';
+import { ViewActions, ViewContext } from '../../../context/view-context/ViewContext';
+import { BridgeContext } from '../context/BridgeContext';
+import { WalletApproveHero } from '../../../components/Hero/WalletApproveHero';
+import { EventTargetContext } from '../../../context/event-target-context/EventTargetContext';
+import { FooterLogo } from '../../../components/Footer/FooterLogo';
+
+export interface ClaimWithdrawalProps {
+  transaction: Transaction,
+}
+export function ClaimWithdrawal({ transaction }: ClaimWithdrawalProps) {
+  const { t } = useTranslation();
+  const { bridgeState: { checkout, tokenBridge, from } } = useContext(BridgeContext);
+  const { viewDispatch } = useContext(ViewContext);
+  const { eventTargetState: { eventTarget } } = useContext(EventTargetContext);
+
+  const { page } = useAnalytics();
+
+  useEffect(() => {
+    page({
+      userJourney: UserJourney.BRIDGE,
+      screen: 'ApproveTransaction',
+    });
+  }, []);
+
+  const [txProcessing, setTxProcessing] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [hasWithdrawError, setHasWithdrawError] = useState(false);
+  const [withdrawalTxn, setWithdrawalTxn] = useState<TransactionRequest | null>();
+
+  const goBack = useCallback(() => {
+    viewDispatch({
+      payload: {
+        type: ViewActions.GO_BACK,
+      },
+    });
+  }, [viewDispatch]);
+
+  useEffect(() => {
+    const getWithdrawalTxn = async () => {
+      if (!tokenBridge) return;
+      // get withdrawal transaction from the token bridge by receipient address and index
+      setLoading(true);
+      try {
+        const flowRateWithdrawTxnResponse = await tokenBridge?.getFlowRateWithdrawTx({
+          recipient: transaction.details.to_address,
+          index: 0, // TODO: update index from transaction when it comes through from backend
+        });
+        setWithdrawalTxn(flowRateWithdrawTxnResponse.unsignedTx);
```

Comment:
save the whole response to state. Then can check the .pendingWithdrawal.canWithdraw property